library_generation/README.md

```diff
@@ -1,235 +1,252 @@
+
+| Name                        | Notes                                                                    |
+|:----------------------------|:-------------------------------------------------------------------------|
+| google-*/                   | Source code generated by gapic-generator-java                            |
+| google-*/pom.xml            | Only be generated if it does not exist                                   |
+| grpc-*/                     | Source code generated by grpc generator, one per each version            |
+| grpc-*/pom.xml              | Only be generated if it does not exist                                   |
+| proto-*/                    | Source code generated by Protobuf default compiler, one per each version |
+| proto-*/pom.xml             | Only be generated if it does not exist                                   |
+| samples/snippets/generated/ | Only be generated if `include_samples` is set to true                    |
+| google-*-bom/pom.xml        | Library BOM, only be generated if it does not exist                      |
+| pom.xml                     | Library parent BOM, only be generated if it does not exist               |
+| .repo-metadata.json         | Always generated from inputs                                             |
+| .OwlBot.yaml                | Only be generated from a template if it does not exist                   |
+| owlbot.py                   | Only be generated from a template if it does not exist                   |
+| README.md                   | Always generated from inputs                                             |
+| gapic-libraries-bom/pom.xml | Always generated from inputs                                             |
+| pom.xml (repo root dir)     | Always generated from inputs                                             |
+| versions.txt                | New entries will be added if they don’t exist                            |
+
+
+## Configuration to generate a repository
+
+There are three levels of parameters in the configuration: repository level,
+library level and GAPIC level.
+
+### Repository level parameters
+
+The repository level parameters define the version of API definition (proto)
+and tools.
+They are shared by library level parameters.
+
+| Name                    | Required | Notes                                        |
+|:------------------------|:--------:|:---------------------------------------------|
+| gapic_generator_version |   Yes    |                                              |
+| protobuf_version        |    No    | inferred from the generator if not specified |
+| grpc_version            |    No    | inferred from the generator if not specified |
+| googleapis-commitish    |   Yes    |                                              |
+| owlbot-cli-image        |   Yes    |                                              |
+| synthtool-commitish     |   Yes    |                                              |
```

Comment:
```suggestion
| googleapis_commitish    |   Yes    |                                              |
| owlbot_cli_image        |   Yes    |                                              |
| synthtool_commitish     |   Yes    |                                              |
| template_excludes       |   Yes    |                                              |
```

Hope you don't mind if I add `template_excludes` which will be merged soon

Reply:
Sure.